### PR TITLE
Add line for EventPlay and related packets

### LINKED ISSUE #343: Feature Request:Log Lines about job Fisher's Mooch opportunities.
When "Mooch" or "Mooch II" can be used in the game, log line 00 will show: "`Mooch to land an even bigger catch!`" or "`You land a fish usable with Mooch II.`" But there are no other corresponding log lines.
Also, I suggest adding log lines corresponding to the "Fish bite" type. For example, using "3" to represent "!!!"
Thank you!

### PR DESCRIPTION
**Pull request by valarnin**
Also add `FishingMsg` to `LineActorControlSelfExtra` as an allowed category.

I elected to include the `Server_MessageHeader` bytes (`0x20` / `32`) in the packet structure as that makes doing the message length check easier/more straightforward.

I plan on making a pass through all our network packet code to clean up the logic in their `MessageReceived` methods, as there are better/more efficient ways to handle what we're doing currently. That'll be in a different PR since I'll be touching basically every custom line.

Closes #343 (mooch status is in the `capture animation` example line, `|1|1|7|` = can use mooch, `|6|1|7|` = can use mooch II).

Could probably use some testing by people who know more about fishing than I do.

As a side effect, this will also expose information about crafting (and likely the other DoH jobs). Here's a copy/paste of part of a DM conversation I had with quisquous about these packets way back on 2023-11-18:

---

Re: crafting chat log lines.
```
4096|2023-11-18T18:07:37.4630000-05:00|EventPlay4|10001234|00000000|000A0001|0000|00040001|00000000|04|00000002|000003F1|00000000|00000001|70215f188f75cf24
00|2023-11-18T18:07:37.0000000-05:00|0842||You begin synthesizing a pair of maple clogs.|49b100a9dd011e9d

4096|2023-11-18T17:34:21.4690000-05:00|EventPlay4|10001234|00000000|000A0001|0000|00040001|00000000|04|00000002|000003F0|00000000|00000001|d38947a8a917602a
00|2023-11-18T17:34:21.0000000-05:00|0842||You begin synthesizing a length of maple lumber.|95cdbb43983a57a3
```
`000003F1`/`1009` = `Recipe.csv` column 1 ID, maps to item ID `3472` = `pair of maple clogs`.
Same thing for length of maple lumber, `3F0` maps through recipe to item.

Using a crafting action, I think there's a flag here that indicates that the craft is complete/failed/etc, there's no other relevant packet sent:
```
4096|2023-11-18T17:34:58.5420000-05:00|EventPlay64|10001234|00000000|000A0001|0000|00040001|00000000|21|00000009|00000000|00000000|00000000|000186A1|00000000|00000001|00000009|000001C8|00000000|00000000|00000001|0000001E|FFFFFFF6|00000001|00000001|00000064|00000016|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|00000000|66cfff98281410c2
00|2023-11-18T17:34:58.0000000-05:00|0842||You use Basic Synthesis  Success!|a0642d780e9ac3af
```

### CODE REVIEW

**Review by wexxlee — COMMENTED** [review]
Continuing the conversation from #343 here.

I'm aligned on the structural decision to integrate `EventStart`/`EventPlay*`/`EventFinish` into a single line - especially to allow future extension - but I'm not convinced all of these varieties need to be written as log lines right now to address the underlying feature request (namely, to expose tug type, fish sizee, and moochability).  Because data beyond that doesn't have an identified use case right now, I think I'd prefer to put some limitations on what we're exposing via this log line until we have additional use cases. 

With that context, here's the changes I'd propose:

- I think we can drop (or comment out, to keep it available for the future) `EventStart` and `EventFinish`.  Definitely recognize that looping animations might have some utility in the future, but since we don't have active examples, I think this is probably extraneous right now.

- In the same vein, I'd suggest dropping all of the `EventPlay` packets **other than** `EventPlay`, `EventPlay4`, and `EventPlay8`.  Those are, I think, the only three needed to address the feature request, and eliminating the others will -- by my count -- avoid the need to update 7 additional opcodes each patch that we aren't actively using.

- I suggest filtering to eventID `150001`.  I'm leery of writing out crafting activity (`0A0001`) or other eventIDs that there isn't a use case for right now.

One other potential issue to address is the variability of params depending on the type of `EventPlay` packet.  While there are only 3 params for `EventPlay`, by my count, there are 6 for `EventPlay4` and 10 for `EventPlay8`, at least in routine fishing activity.  

This is admittedly a cactbot problem and not something other consumers necessarily share, but a variable # of params makes capture regex very difficult -- particularly true since fish size and moochability are in the extra params on `EventPlay4` and `EventPlay8` respectively, and those params are not present in `EventPlay`.  Given that cactbot has retired the `fisher` module, I don't anticipate we'd do much other than add a basic entry to `netlog_defs` to support the new line type, but unless we do some refactoring, everything beyond `param3` will not be capturable.

I think there's at least three options to address this:

1. Write out a fixed # of params, and zero-fill the missing params for `EventPlay` and `EventPlay4`. This is easiest from the cactbot perspective, but would require a line version update if we later add additional `EventPlay*` packet types or if there are additional variable params for non-fishing events.
2. Leave it up to consumers (cactbot) to handle.  Given the `fisher` module has been retired, I don't see refactoring `parseHelper`/`buildRegex` as a priority, so this probably just means that anything beyond `param3` (including fish size/moochability) will not be captured.
3. Make this line specific to fishing activity rather than a generic `EventPlay` line, and abstract the output specifically for fishing, rather than raw param data.  E.g.,:
```
275|<ts>|<actorId>|<scene>|<tug/fish type>|<Moochable>
275|<ts>|10001234|1|113|0
275|<ts>|10001234|2|11B|0
275|<ts>|10001234|3|111|0
275|<ts>|10001234|4|10F|1 (or 6)
275|<ts>|10001234|5|124|0
275|<ts>|10001234|6|11D|0
```

While this is cleanest from a readability/parsing standpoint, it means more burden on OP to process the lines (and will require maintainance if fishing data format ever changes).  It also means a line version update for future feature requests that depend on other params not exposed -- so this is probably not my favorite option, but worth flagging.

---------------

Sort of an aside, but fwiw, here's what I have for mapping the `scene` to specific fishing actions/animations.  Additional log line examples that could be added to the comments (for the missing 'cast' example) or to `LogGuide`:

1 CAST (`EventPlay`)
```275|2024-05-27T05:28:09.5930000-07:00|1|10FFFFFF|0|150001|1|40001|0|2|113|```

2 REEL-IN (`EventPlay`) - sent if 'Quit' used, after an ignored BITE, or after HOOK
```275|2024-05-27T05:28:33.3550000-07:00|1|10FFFFFF|0|150001|2|40001|0|2|11B|```

3 PUT-AWAY (`EventPlay`) - sent after `Quit` used; follows REEL-IN (if used while Cast active)
```275|2024-05-27T05:28:49.8000000-07:00|1|10FFFFFF|0|150001|3|40001|0|1|111|```

4 RESULT (`EventPlay8`) - sent after HOOK, or after REEL-IN following an ignored BITE.
```
// ignored bite
275|2024-05-27T05:28:34.6900000-07:00|8|10FFFFFF|0|150001|4|40001|0|5|10F|0|0|0|0|0|0|0|
// caught
275|2024-05-27T05:29:11.9990000-07:00|8|10FFFFFF|0|150001|4|40001|0|5|10F|0|0|1|7|0|0|0|
// don't yet have an example of a hook-escape
```
5 BITE (`EventPlay`) - sent on a bite, and contains headmarker in `param3`:
`275|2024-05-27T05:29:04.4280000-07:00|1|10FFFFFF|0|150001|5|40001|0|1|124|`

6 HOOK (`EventPlay4`) - sent if 'Hook' (or variations) action used, contains combined headmarker+size in `param4`.
`275|2024-05-27T05:29:05.4940000-07:00|4|10FFFFFF|0|150001|6|40001|0|4|11C|11D|10F|0|`

**Comment by valarnin** [comment]
> Because data beyond that doesn't have an identified use case right now, I think I'd prefer to put some limitations on what we're exposing via this log line until we have additional use cases.

There are use cases that were solved in other/more complicated ways elsewhere, e.g. the vast majority of the cactbot jobs crafter stuff using chat log lines currently.

There's also a question of "if people writing triggers never see these packets, how are they supposed to know they can use them"?

I'm not aware of any `Event*` packets in battle content, so there's not really a concern here from a too-many-log-lines-generated standpoint.

---

> * I think we can drop (or comment out, to keep it available for the future) `EventStart` and `EventFinish`.  Definitely recognize that looping animations might have some utility in the future, but since we don't have active examples, I think this is probably extraneous right now.

I'm fine with dropping these. I only included them as they're defined and handled along the same lines as the `EventPlay*` packets. The only place I'm aware of them being used is on zone in and after killing a boss/triggering the "victory" cutscene. They might exist elsewhere in content that I don't frequently run, but I'm not willing to dig through almost 20gb of log files trying to find those rare instances.

---

> * In the same vein, I'd suggest dropping all of the EventPlay packets other than EventPlay, EventPlay4, and EventPlay8. Those are, I think, the only three needed to address the feature request, and eliminating the others will -- by my count -- avoid the need to update 7 additional opcodes each patch that we aren't actively using.

This I'm less fine with. See above re: the continued removal of reliance on chat log lines for `cactbot`.

As far as I'm aware, updating opcodes is an all-or-nothing thing. It either works for all the opcodes, or it's broken for all the opcodes and needs to be fixed regardless of how many there are.

I'm also working on the side on a project that can live under the OverlayPlugin org which is written in C# and can extract opcodes from the game exe directly, without the need for a proper decompilation via IDA or Ghidra, though this is still in early stages (it currently breaks due to sig scan matching on the raw exe, but works when I target the exe in memory, need to do more research).

---

> * I suggest filtering to eventID 150001. I'm leery of writing out crafting activity (0A0001) or other eventIDs that there isn't a use case for right now.

I'm not sure I understand where this reluctance is coming from, can you elaborate?

---

> One other potential issue to address is the variability of params depending on the type of `EventPlay` packet. While there are only 3 params for `EventPlay`, by my count, there are 6 for `EventPlay4` and 10 for `EventPlay8`, at least in routine fishing activity.
> 
> This is admittedly a cactbot problem and not something other consumers necessarily share, but a variable # of params makes capture regex very difficult -- particularly true since fish size and moochability are in the extra params on `EventPlay4` and `EventPlay8` respectively, and those params are not present in `EventPlay`. Given that cactbot has retired the `fisher` module, I don't anticipate we'd do much other than add a basic entry to `netlog_defs` to support the new line type, but unless we do some refactoring, everything beyond `param3` will not be capturable.

Parameter count is included in the log line. `cactbot` can treat this line similarly to how it treats `gameLog` currently, e.g. in NetRegexes have a `static eventPlay` which matches any `eventPlay` line, and then `static eventPlay1` that matches an `eventPlay` line with `1` in the `parameterCount` field, etc.

If the `parameterCount` field states `8`, then there will always be 8 parameters at the end. The actual definition in `netlog_defs.ts` will list all the parameters possible, but mark param2+ as optional so they're not automatically included in the capture regex. Or possibly the repeating fields logic can be leveraged.

**Comment by wexxlee** [comment]
> > Because data beyond that doesn't have an identified use case right now, I think I'd prefer to put some limitations on what we're exposing via this log line until we have additional use cases.
> 
> There are use cases that were solved in other/more complicated ways elsewhere, e.g. the vast majority of the cactbot jobs crafter stuff using chat log lines currently.

I guess my point, though, is that they have been solved, so unless there's another reason to switch to a different solution or it's likely to be a future/recurring issue and would benefit from a new line type, the benefit isn't as strong.  (The `jobs` module reliance on chat log lines is, I think, one such reason -- more on that below.)

> There's also a question of "if people writing triggers never see these packets, how are they supposed to know they can use them"?

I'm not sure I necessarily agree with this.  The logical outgrowth of that approach is that we should include all conceivable data we can identify, regardless of existing utility, under the theory that it may someday be useful for some purpose.  I think the more functional approach to trigger writing, when looking at a mechanic that needs a trigger, is asking "what info from the game do I need to write a trigger for this mechanic" - and if that info isn't exposed (or isn't timely exposed) in existing log lines but is still available in some form to the player (spawned objects, animations, speech bubbles, etc.), we then figure out how to source it and include a log line for it.

Perhaps this is a better conversation to have in #81, but I guess I'm looking to align on the limiting principle(s) for when we do or don't add a new log line, because there's a wealth of packet data we could start turning into lines.  Is it just "spaminess"?  (I agree that's one consideration, but that feels too subjective to be the sole criteria?).  Some of the discussion in #81 focused around limiting lines to output that's useful for triggers, so that seems perhaps like a reasonable starting point: new log lines should be confined to data where we have an identified need or downstream request, and should be reasonably limited to data that is/can be useful in some known or planned application.  I guess I'm hesitating here because I'm not sure what the limiting principle is (or should be).  Not that I necessarily have to agree with it -- I'm just not confident what it is.

(That all said, this is just my $0.02.  100% open to more discussion/persuasion.  Just trying to articulate what my current hesitation is.)


----


> I'm not aware of any `Event*` packets in battle content, so there's not really a concern here from a too-many-log-lines-generated standpoint.

I've only done very limited testing, but I found `Event*` lines being written in both instanced and non-instanced content:

```
// based on timing, I think this likely has to do with Tribal Quest content
// nts - are params on EventStart & EventFinish packets not being trimmed?
275|2024-05-27T17:39:25.2380000-07:00|S|00FF0000|00000001|000111C7|01|00|00000000|
275|2024-05-27T17:39:25.2380000-07:00|1|00FF0000|1|111C7|9|2000|0|0|0|
275|2024-05-27T17:39:34.1030000-07:00|F|00FF0001|01|01|00000000|
275|2024-05-27T17:39:36.5970000-07:00|S|00FF0000|00000001|000111C8|01|00|00000000|
275|2024-05-27T17:39:36.5970000-07:00|1|00FF0000|1|111C8|7|2000|0|0|0|
275|2024-05-27T17:39:39.0040000-07:00|1|00FF0000|1|111C8|8|2000|0|0|0|
275|2024-05-27T17:39:44.1680000-07:00|F|00FF0002|01|01|00000000|
275|2024-05-27T17:39:50.0040000-07:00|S|00FF0000|00000001|000111C9|01|00|00000000|
275|2024-05-27T17:39:50.0040000-07:00|1|00FF0000|1|111C9|A|2000|0|0|0|
275|2024-05-27T17:39:52.2750000-07:00|1|00FF0000|1|111C9|B|2000|0|0|0|
```

```
// log of all Event* packets sent during a run of Labyrinth of the Ancients
275|2024-05-27T23:00:12.2280000-07:00|S|10FF1111|00000000|80037531|0F|01|000000AE|
275|2024-05-27T23:00:12.2280000-07:00|16|10FF1111|0|80037531|1|C53A401|0|C|B4|0|0|5|62671565|0|0|0|0|0|1C20|80|0|0|0|0|
275|2024-05-27T23:00:15.3020000-07:00|F|80037531|0F|01|000000AE|
275|2024-05-27T23:05:21.4510000-07:00|S|40004621|00000000|000F0085|01|00|00000000|
275|2024-05-27T23:05:21.4510000-07:00|1|40004621|0|F0085|1|2001|0|0|0|
275|2024-05-27T23:05:22.4300000-07:00|F|000F0085|01|01|00000000|
275|2024-05-27T23:09:00.6870000-07:00|S|40004614|00000000|000F0098|01|00|00000000|
275|2024-05-27T23:09:00.6870000-07:00|1|40004614|0|F0098|1|2001|0|0|0|
275|2024-05-27T23:09:01.6240000-07:00|F|000F0098|01|01|00000000|
275|2024-05-27T23:13:23.5780000-07:00|S|10FF1111|00000000|80037531|0D|01|00000001|
275|2024-05-27T23:13:23.5780000-07:00|8|1093A385|0|80037531|3|8502403|0|5|19A|1|B5|0|0|0|0|0|
275|2024-05-27T23:13:25.1690000-07:00|F|80037531|0D|01|00000001|
275|2024-05-27T23:13:27.7980000-07:00|S|40004605|00000000|000F012E|01|00|00000000|
275|2024-05-27T23:13:27.7980000-07:00|1|40004605|0|F012E|1|2001|0|0|0|
275|2024-05-27T23:13:28.6450000-07:00|F|000F012E|01|01|00000000|
275|2024-05-27T23:16:00.3300000-07:00|S|10FF1111|00000000|80037531|0D|01|00000001|
275|2024-05-27T23:16:00.3300000-07:00|8|10FF1111|0|80037531|3|8502403|0|5|19B|1|12|0|0|0|0|0|
275|2024-05-27T23:16:01.9120000-07:00|F|80037531|0D|01|00000001|
275|2024-05-27T23:16:08.5540000-07:00|S|400045F4|00000000|000F0005|01|00|00000000|
275|2024-05-27T23:16:08.5540000-07:00|1|400045F4|0|F0005|1|2001|0|0|0|
275|2024-05-27T23:16:11.5340000-07:00|F|000F0005|01|01|00000000|
```

I found others from the last day or so of ordinary existence in game, but I haven't dug in to unpack them yet.

----

> > * In the same vein, I'd suggest dropping all of the EventPlay packets other than EventPlay, EventPlay4, and EventPlay8. Those are, I think, the only three needed to address the feature request, and eliminating the others will -- by my count -- avoid the need to update 7 additional opcodes each patch that we aren't actively using.
> 
> This I'm less fine with. See above re: the continued removal of reliance on chat log lines for `cactbot`.

I haven't used the `jobs` module for crafter, so I may be missing some nuances, but as far as I can tell, there are 8 translated regexes that relate to crafting, and they are used only to show if crafting is currently underway (to decide whether to show the CP bar).  It looks like all of these can be determined from `EventPlay4` or existing 0x29 lines:

```
// trialCraftingStart
// EventPlay4: param1 = 2 & param2 = 885C -> Recipe id 34908 -> Rarefied Sykon Bavarois
275|<ts>|S|10FF0001|00000000|000A0001|06|00|00000000|
275|<ts>|4|10FF0001|0|A0001|0|40001|0|4|1|0|4B97D366|400017|
275|<ts>|4|10FF0001|0|A0001|0|40001|0|4|2|885C|0|1|
275|<ts>|4|10FF0001|0|A0001|0|40001|0|4|3|0|0|0|

// trialCraftingCancel
// EventPlay4: param1 = 6
275|<ts>|4|10FF0001|0|A0001|0|40001|0|4|6|0|A|0|
// the EventFinish line is missing actorId + flags for some reason?  This will definitely cause regex issues...
275|<ts>|F|000A0001|06|01|00000000|

// trialCraftingFinish
// SystemLogMessage: instance = A0001; id = 170D -> LogMessage 5901 -> "Trial synthesis complete!"
// (although the actual log message displayed appears to be 0x170E)
// param1 = 8F12 -> Item id 36626 -> serving of rarefied sykon bavarois
41|<ts>|A0001|170D|02|8F12|00|484230e3202c7b33
// EventPlay4: param1 = 4
275|<ts>|4|10FF0001|0|A0001|0|40001|0|4|4|0|0|0|

// trialCraftingFail
// SystemLogMesage: instance = A0001; id = 170F -> LogMessage 5903 -> "Trial synthesis failed..."
// (although the actual log message displayed appears to be 0x1710)
// param1 = 8F12 -> Item id 36626 -> serving of rarefied sykon bavarois
41|<ts>|A0001|170F|02|8F12|00|46e93873211f6d6c
// EventPlay4: param1 = 4
275|<ts>|4|10FF0001|0|A0001|0|40001|0|4|4|0|0|0|

// craftingStart + craftingCancel + craftingFinish + craftingFail
// same as trialCrafting* - there doesn't seem to be a way to tell from Event* packets whether this is trial or actual
```

TLDR: It seems like cactbot `jobs`'s dependencies on chatlog regex can be replaced by `EventPlay4`, using param1 ("2" = start, with param2 as the recipeId; "6" = cancel, or "4" = fail/success), and if the actual completed recipe is necessary for a fail/success, the 0x29 line can be used instead.  While it might be possible to determine fail/success from `EventPlay64` -- which seems to be sent after every crafting action used (at first glance, it looks like param13 corresponds to durability, so param13 = 0 would be the end of a craft, success or fail) -- I think some combination of `EventPlay4` and `SystemLogMessage` seem like a better alternative.

----

> As far as I'm aware, updating opcodes is an all-or-nothing thing. It either works for all the opcodes, or it's broken for all the opcodes and needs to be fixed regardless of how many there are.
> 
> I'm also working on the side on a project that can live under the OverlayPlugin org which is written in C# and can extract opcodes from the game exe directly, without the need for a proper decompilation via IDA or Ghidra, though this is still in early stages (it currently breaks due to sig scan matching on the raw exe, but works when I target the exe in memory, need to do more research).

Sounds like a great project!  Yeah, looking back at Makar's [comment](https://github.com/OverlayPlugin/OverlayPlugin/issues/276#issuecomment-1817732008), it does seem like an all-or-nothing thing for how they do opcode updates for global.  Not sure if that's the same for Ra and ShadyWhite (or anyone else) who update the kr/cn opcodes though.  Either way, I agree this isn't a blocker -- more just mentioning it.

----

> > * I suggest filtering to eventID 150001. I'm leery of writing out crafting activity (0A0001) or other eventIDs that there isn't a use case for right now.
> 
> I'm not sure I understand where this reluctance is coming from, can you elaborate?

I think I probably addressed this above, but this stems from my overall hesitation about scope of the PR vs. scope of the feature request.  Looking more at the lines generated from crafting activity, it's pretty minimal and given the discussion above about the `jobs` module, I'll walk that part back.  I guess I just don't really feel like I have a handle yet on what the true packet volume really is, since there clearly is some overworld/battle packets being generated too.  That said, I'm fine parking this suggestion; we can revisit if there is a noise issue later.

----

> > One other potential issue to address is the variability of params depending on the type of `EventPlay` packet. [...]
> Parameter count is included in the log line. `cactbot` can treat this line similarly to how it treats `gameLog` currently, e.g. in NetRegexes have a `static eventPlay` which matches any `eventPlay` line, and then `static eventPlay1` that matches an `eventPlay` line with `1` in the `parameterCount` field, etc.
> 
> If the `parameterCount` field states `8`, then there will always be 8 parameters at the end. The actual definition in `netlog_defs.ts` will list all the parameters possible, but mark param2+ as optional so they're not automatically included in the capture regex. Or possibly the repeating fields logic can be leveraged.

Yeah - there's a couple of options for doing it, but I guess my point was that it'll require a little work beyond just a logdef update.  That being said, this does sort of spotlight the issue of including EventPlay16-255.  I'm not a fan of listing out all 255 param fields in the log definition 😶.  Repeating fields might work, but I think it might require some rejiggering of the key/value/pair logic?  Either way, it's some work.  Much less so if we're only dealing with 8 params; in that case, I think some static Netregexes would work fine.

**Comment by wexxlee** [comment]
> My general thoughts and process are (extremely vaguely) as follows:
> 
> 1. Identify a need for a new log line
> 2. Determine the source of the data
> 3. Verify that the new log line can solve the problem
> 4. Check for frequency of the packet
> 5. Determine if the new log line should be limited in scope depending on the frequency of the packet
> 
> In this case, the `Event*` packets are all similarly structured and used for similar things, so it made sense to group them together (the same way base `FFXIV_ACT_Plugin` groups together the `ActionEffect1 .. ActionEffect32` packets). And at that point, there was no reason to not include the remaining packets, since their volume is basically non-existent and the infrastructure to handle them was already being added anyways.

That makes sense, and I think we're on the same page.  I suppose packet frequency will always be a subjective thing, but I think I'm content to burn that bridge when we come to it.

> I'm fine with either leaving the larger packets out or including them, ultimately it makes no difference to me. I just want to make sure we're on the same page as far as thought processes and such are concerned.

I think I have a slight preference at this point for filtering out EventPlay16+ for some of the reasons already covered.  As I've been thinking about it more, I'm also concerned about the data volume from crafting (especially with a new expac around the corner) - we'll already be writing a handful of EventStart, EventFinish, and EventPlay[1,4] packets for each craft, and the line length and volume of EventPlay64 packets for each crafting action will, I think, add a fair bit of clutter to log files.

I have a couple of suggestions on line format/params that I'll tack on in a second, but wanted to flag something else -- at the moment, I'm not able to test this build because it's causing ACT to get stuck during init.  I've spent some time debugging and traced the cause to the fact that there are no KR/CN opcodes for the various Event* packets, and as a result, `RegionalizedPacketHelper.CreateFromOpcodeConfig` is attempting to fetch the opcodes from GH.  When `SaveRemoteOpcodesToConfig` calls `HttpClientWrapper.Get`, the process seems to wait indefinitely and ACT never completes loading.

I'm not sure about the underlying issue, since I have no problems accessing the URL directly.  I tried adding a cancellation token to `HttpClientWrapper.Get`, but that didn't fix it.  And I'm not sure if this is something on my end -- are you able to repro?  I won't have time to poke back at it until later in the week, but thought I'd mention it now in case you have a chance to look.

File: OverlayPlugin.Core/NetworkProcessors/LineEventStartPlayFinish.cs [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+            uint eventID;
+            byte param1;
+            byte param2;
+            ushort padding;
+            uint param3;
+            uint padding1;
+
+            public string ToString(long epoch, uint ActorID)
+            {
+                return $"S|{actorID:X8}|{unknown1:X8}|{eventID:X8}|{param1:X2}|{param2:X2}|{param3:X8}";
```
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineEventStartPlayFinish.cs:209]: One thing that I think may be problematic from a consumer standpoint is that the "static" (non-eventParam) fields are different between EventPlay*, EventStart, and EventFinish, which makes an accurate log def a little more challenging.  To standardize, what about adding (either empty-string or zero-filled) placeholders for `scene` and `flags`?

Also, just in terms of formatting, could we change `unknown1`, `eventID`, and `param1-3` to `:X` so they match `EventPlay*`?

File: OverlayPlugin.Core/NetworkProcessors/LineEventStartPlayFinish.cs [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+            uint eventID;
+            byte param1;
+            byte param2;
+            ushort padding;
+            uint param3;
+            uint padding1;
+
+            public string ToString(long epoch, uint ActorID)
+            {
+                return $"F|{eventID:X8}|{param1:X2}|{param2:X2}|{param3:X8}";
```
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineEventStartPlayFinish.cs:225]: Same comments as `EventStart` re: missing fields and `:X` formatting.  Kind of odd too that `actorID` isn't in the packet - perhaps we could just zero-fill that?

File: OverlayPlugin.Core/NetworkProcessors/LineEventStartPlayFinish.cs [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+        internal unsafe struct EventPlayBase
+        {
+            uint actorID;
+            uint unknown1;
+            uint eventID;
+            ushort scene;
+            ushort padding;
+            uint flags;
+            uint param3;
+            byte param4;
```
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineEventStartPlayFinish.cs:34]: Any particular reason these are `param3` and `param4` as opposed to `1` and `2`?  (ClientStructs thing?)  It just caught my eye as a little odd that we start the numbering at 3, but start at 1 for `EventStart`/`EventFinish`.  (Yes, this is internal and probably immaterial since we can relabel the fields in `netlog_defs`, so mostly just curious - not necessarily a change request.

**Review by wexxlee — COMMENTED** [review]
A few minor suggestions on line formatting.

File: OverlayPlugin.Core/NetworkProcessors/LineEventStartPlayFinish.cs (hunk 3 above)
Comment (valarnin) [reply OverlayPlugin.Core/NetworkProcessors/LineEventStartPlayFinish.cs:34]: It starts at `3` because originally `scene/padding` were `param1`, and `flags` was `param2`. The joys of trying to figure these things out.

**Comment by valarnin** [comment]
> I have a couple of suggestions on line format/params that I'll tack on in a second, but wanted to flag something else -- at the moment, I'm not able to test this build because it's causing ACT to get stuck during init. I've spent some time debugging and traced the cause to the fact that there are no KR/CN opcodes for the various Event* packets, and as a result, `RegionalizedPacketHelper.CreateFromOpcodeConfig` is attempting to fetch the opcodes from GH. When `SaveRemoteOpcodesToConfig` calls `HttpClientWrapper.Get`, the process seems to wait indefinitely and ACT never completes loading.
> 
> I'm not sure about the underlying issue, since I have no problems accessing the URL directly. I tried adding a cancellation token to `HttpClientWrapper.Get`, but that didn't fix it. And I'm not sure if this is something on my end -- are you able to repro? I won't have time to poke back at it until later in the week, but thought I'd mention it now in case you have a chance to look.

I've had it happen sometimes too, but not yet with a debugger attached. It's possible that the `Get` is throwing an exception that's being caught and silenced up the chain somewhere.

I think this has always been a potential problem, and we're just seeing it when testing this because it's always reaching out to GH for the opcodes file since the values are missing.

The underlying logic we use for any HTTP request needs a bit of an overhaul to be honest, because right now there's a lot of uses that call the `Get` method on the UI thread and therefore can block the UI thread indefinitely. A pass through the code to ensure that we're not executing these methods directly on the UI thread would probably be a good idea in general. Maybe some code in `#if DEBUG` in the `HttpClientWrapper` `Get` method that dumps an error with stack trace to the log file if it detects it's on the UI thread?

In this particular case we're loading `FFXIVCustomLogLines` within the UI thread in `PluginMain`, so when it fails it's blocking the UI thread indefinitely. The whole `PluginMain` class is a bit of a mess at this point and could use some general cleanup too...

**Comment by valarnin** [comment]
The tl;dr on the "fix" for the freeze is that ultimately, running async/await on the UI thread is indeed a bad idea and will completely deadlock the UI thread sometimes due to which thread gets assigned to process the async function. So we tell the `Task` to always pass context back to the same thread that called it, to avoid the issue.

This should still be fixed properly by not running `HttpClient` async stuff on the UI thread to begin with, but that's pending a rewrite of how `PluginLoader` and `PluginMain` actually initialize the plugin.

**Comment by valarnin** [comment]
I did some more digging through my log files, and across log files back to 2024-01-08, I had this many of each packet:

| Count | Packet |
| -- | -- |
| 1738 | EventPlay |
| 359 | EventPlay4 |
| 531 | EventPlay8 |
| 3687 | EventPlay16 |
| 15 | EventPlay32 |
| 10 | EventPlay64 |
| 0 | EventPlay128 |
| 13 | EventPlay255 |
| 2612 | EventStart |
| 2578 | EventFinish |

- For `EventPlay255`, every single instance was me purchasing a stack of Dark Matter from a Junkmonger. 
- I'm around other players doing crafting fairly often, and their crafting actions never seem to trigger any `Event*` packets.
- `0` instances of `EventPlay128`

Given the above, I propose the following:

- Split `EventStart` and `EventEnd` off to their own line(s), potentially as a separate PR
- Remove `EventPlay128` and `EventPlay256`, as they appear to have absolutely no potential use currently
- Keep `EventPlay32` and `EventPlay64`, as they have potential use for crafting and they're only sent for the current player
- (possibly) Add a guard condition to `EventPlay*` packets to not print them for players that aren't the current player. Continue to allow them for non-player actors.
